crypto/evp/p_lib.c

```diff
@@ -106,43 +126,163 @@ int EVP_PKEY_copy_parameters(EVP_PKEY *to, const EVP_PKEY *from)
         return 0;
     }
 
-    if (from->ameth && from->ameth->param_copy)
+    /*
+     * If |from| is provided, we upgrade |to| to be provided as well.
+     * This drops the legacy key from |to|.
+     * evp_pkey_upgrade_to_provider() checks if |to| is already provided,
+     * we don't need to do that here.
+     */
+    if (from->keymgmt != NULL) {
+        EVP_KEYMGMT *tmp_keymgmt = from->keymgmt;
+
+        evp_pkey_upgrade_to_provider(to, NULL, &tmp_keymgmt, NULL);
```

Comment:
Can this function not fail?

Reply:
Yes, but that shows in `to->keymgmt`, which remains NULL in that case.

Reply:
I prefer not relying on side effects of function internals. The function could change over time (it appears to be changing quite a lot at the moment :)).

Reply:
Come to think of it, I misremembered this.  Check added.